hdinsight-with-edge-node/azuredeploy.json

```diff
@@ -0,0 +1,402 @@
+            }
+        },
+        "clusterLoginPassword": {
+            "type": "securestring",
+            "metadata": { 
+                "description": "The password for the cluster login."
+            }
+        },
+        "sshUserName": {
+            "type": "string",
+            "defaultValue": "hdiuser",
+            "metadata": { 
+                "description": "These credentials can be used to remotely access the cluster and the edge node virtual machine."
+            }
+        },
+        "sshPassword": {
+            "type": "securestring",
+            "metadata": { 
+                "description": "The password for the ssh user."
+            }
+        },
+        "clusterStorageAccountName": {
+            "type": "string",
+            "metadata": { 
+                "description": "The name of the storage account to be created and be used as the cluster's storage."
+            }
+        },
+        "clusterWorkerNodeCount": {
+            "type": "int",
+            "defaultValue": 4,
+            "metadata": { 
+                "description": "The number of nodes in the HDInsight cluster."
+            }
+        },
+        "edgeNodeVhdStorageName": {
+            "type": "string",
+            "metadata": { 
+                "description": "The name of the storage account to be created and used to store the vhd of the edge node virtual machine."
+            }
+        },
+        "edgeNodeVirtualMachineName": {
+            "type": "string",
+            "metadata": { 
+                "description": "The name of the edge node virtual machine."
+            }
+        },
+        "appInstallScriptUri": {
+            "type": "string",
+            "defaultValue": "https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/hdinsight-with-edge-node/connectToWASB.sh",
+            "metadata": { 
+                "description": "A public http(s) uri that points to a script that will install an app on the edge node."
+            }
+        }
+    },
+    "variables": {
+        "edgeNodeStorageAccountContainerName": "vhds",
+        "edgeNodeImagePublisher": "Canonical",
+        "edgeNodeImageOffer": "UbuntuServer",
+        "edgeNodeVmSize": "Standard_D1",
+        "edgeNodeExtensionName": "edgeNodeSetup",
+        "edgeNodeOSDiskName": "[parameters('edgeNodeVirtualMachineName')]",
+        "edgeNodeNicName": "[concat(parameters('edgeNodeVirtualMachineName'), 'nic')]",
+        "edgeNodePublicIpName": "[concat(variables('edgeNodeNicName'), 'publicip')]",
+        "edgeNodeNicSSHSGName": "[concat(variables('edgeNodeNicName'), 'ssh')]"
+    },
+    "resources": [
+        {
+            "name": "[parameters('vnetName')]",
+            "type": "Microsoft.Network/virtualNetworks",
+            "location": "[parameters('location')]",
+            "apiVersion": "2015-05-01-preview",
```

Comment:
Would be a good idea to make the api version a variable so it's easier to move to newer api versions